googleapiclient/discovery.py

```diff
@@ -1110,19 +1125,36 @@ def methodResource(self):
                                method.__get__(self, self.__class__))
 
   def _add_next_methods(self, resourceDesc, schema):
-    # Add _next() methods
-    # Look for response bodies in schema that contain nextPageToken, and methods
-    # that take a pageToken parameter.
+    # Add _next() methods iff one of the names 'pageToken' or 'nextPageToken'
+    # occurs among the fields of both the method's response type and either the
+    # method's request (query parameters) or request body.
     if 'methods' in resourceDesc:
       for methodName, methodDesc in six.iteritems(resourceDesc['methods']):
-        if 'response' in methodDesc:
-          responseSchema = methodDesc['response']
-          if '$ref' in responseSchema:
-            responseSchema = schema.get(responseSchema['$ref'])
-          hasNextPageToken = 'nextPageToken' in responseSchema.get('properties',
-                                                                   {})
-          hasPageToken = 'pageToken' in methodDesc.get('parameters', {})
-          if hasNextPageToken and hasPageToken:
-            fixedMethodName, method = createNextMethod(methodName + '_next')
+        nextPageToken = pageTokenName(
```

Comment:
Since you're here, why not fix this code to not be quite so nested? Early return clauses would make this much more readable.